src/client/core/utils/dom.js

```diff
@@ -364,28 +367,78 @@ export function getElementDescription (el) {
+    if (!isRadioButtonElement(sourceElement))
+        return elements;
+
+    const isArrowNavigationAllowed                  = !skipRadioGroups && sourceElement.name !== '';
+    const isArrowNavigationBetweenNonamesAllowed    = !skipRadioGroups && !sourceElement.name && browserUtils.isChrome;
+    const isArrowNavigationBetweenNonamesDisallowed = !skipRadioGroups && !sourceElement.name && !browserUtils.isChrome;
+
+    elements = arrayUtils.filter(elements, item => {
+        const isRadioItem = isRadioButtonElement(item);
+
+        if (isArrowNavigationAllowed)
+            return isRadioItem && item.name === sourceElement.name;
+
+        if (isArrowNavigationBetweenNonamesAllowed)
+            return isRadioItem && !item.name;
+
+        if (isArrowNavigationBetweenNonamesDisallowed)
+            return item === sourceElement;
+
+        // NOTE: in all browsers except Mozilla and Opera focus sets on one radio set from group only.
+        // in Mozilla and Opera focus sets on any radio set.
+        if (sourceElement.name !== '' && !browserUtils.isFirefox)
+            return !item.name || item === sourceElement || item.name !== sourceElement.name;
+
+        return true;
+    });
+
+    return elements;
+}
+
+function correctFocusableElement (elements, element, skipRadioGroups) {
+    const isNotCheckedRadioButtonElement = isRadioButtonElement(element) && element.name && !element.checked;
+
+    if (!skipRadioGroups || !isNotCheckedRadioButtonElement)
+        return element;
+
+    const checkedRadioButtonElementWithSameName = arrayUtils.find(elements, el => {
+        return isRadioButtonElement(el) && el.name === element.name && el.checked;
+    });
+
+    return checkedRadioButtonElementWithSameName || element;
+}
+
+export function getNextFocusableElement (element, reverse, skipRadioGroups) {
+    const offset     = reverse ? -1 : 1;
     let allFocusable = sortElementsByFocusingIndex(getFocusableElements(findDocument(element)));
 
-    //NOTE: in all browsers except Mozilla and Opera focus sets on one radio set from group only.
-    // in Mozilla and Opera focus sets on any radio set.
-    if (isInputElement(element) && element.type === 'radio' && element.name !== '' && !browserUtils.isFirefox) {
-        allFocusable = arrayUtils.filter(allFocusable, item => {
-            return !item.name || item === element || item.name !== element.name;
-        });
-    }
+    allFocusable = filterFocusableElements(allFocusable, element, skipRadioGroups);
 
+    const isRadioInput         = isRadioButtonElement(element);
     const currentIndex         = arrayUtils.indexOf(allFocusable, element);
     const isLastElementFocused = reverse ? currentIndex === 0 : currentIndex === allFocusable.length - 1;
 
     if (isLastElementFocused)
-        return document.body;
+        return skipRadioGroups || !isRadioInput ? document.body : allFocusable[allFocusable.length - 1 - currentIndex];
```

Comment:
What if the `element` is the last radio element in a group, and some next elements have the same name (`correctFocusableElements` case)?

Reply:
`correctFocusableElements` is case when you press tab and focus radio button which is not checked. In this case we need focus other radio button with the first one's name.